### PR TITLE
[ESKL-130/show-opponent-status] - Shows a toast when opponent finishes

### PR DESCRIPTION
**Pull request by nnelas**
**What does this PR do?**

- Shows a toast when opponent finishes the match

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] 2048/2048/src/main/java/com/appcoins/eskills2048/MainActivity.java

**How should this be manually tested?**

Enter a new game with 2 devices. End game in one of them. On the other one you should see a Toast appearing saying your opponent has finished the game.

**What are the relevant tickets?**

  Tickets related to this pull-request: [ESKL-130](https://aptoide.atlassian.net/browse/ESKL-130)

**Questions:**

N/A

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: 2048/2048/src/main/java/com/appcoins/eskills2048/MainActivity.java [hunk 1]
```diff
@@ -46,7 +49,8 @@ public static Intent newIntent(Context context, String userId, String walletAddr
     super.onCreate(savedInstanceState);
     view = new MainView(this,
         new MainGameViewModel(setScoreUseCase, setFinalScoreUseCase, buildViewModelData(),
-            getRoomUseCase, setGameStatusLocallyUseCase), userDetailsHelper, userDataStorage);
+            getRoomUseCase, setGameStatusLocallyUseCase, new NotifyOpponentFinishedUseCase(this)),
```
Comment (Trinkes) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/MainActivity.java:52]:
```suggestion
            getRoomUseCase, setGameStatusLocallyUseCase, new NotifyOpponentFinishedUseCase(opponent -> view.game.onOpponentFinished(opponent))),
```

File: 2048/2048/src/main/java/com/appcoins/eskills2048/vm/MainGameViewModel.java [hunk 2]
```diff
@@ -55,4 +60,8 @@ public void setGameStatus(Tile[][] field, long score) {
   public LocalGameStatus getGameStatus() {
     return data.getLocalGameStatus();
   }
+
+  public void notify(User opponent) {
```
Comment (Trinkes) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/vm/MainGameViewModel.java:64]:
```suggestion

```

File: 2048/2048/src/main/java/com/appcoins/eskills2048/vm/MainGameViewModel.java [hunk 3]
```diff
@@ -55,4 +60,8 @@ public void setGameStatus(Tile[][] field, long score) {
...
+    notifyOpponentFinishedUseCase.notify(opponent);
```
Comment (Trinkes) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/vm/MainGameViewModel.java:65]:
```suggestion

```

File: 2048/2048/src/main/java/com/appcoins/eskills2048/vm/MainGameViewModel.java [hunk 4]
```diff
@@ -55,4 +60,8 @@ public void setGameStatus(Tile[][] field, long score) {
...
+  }
```
Comment (Trinkes) [inline 2048/2048/src/main/java/com/appcoins/eskills2048/vm/MainGameViewModel.java:66]:
```suggestion

```

**Review by Trinkes — CHANGES_REQUESTED** [review]

File: 2048/2048/src/main/java/com/appcoins/eskills2048/usecase/NotifyOpponentFinishedUseCase.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package com.appcoins.eskills2048.usecase;
+
+import com.appcoins.eskills2048.OpponentStatusListener;
+import com.appcoins.eskills2048.model.User;
+import com.appcoins.eskills2048.model.UserStatus;
+import java.util.ArrayList;
+import java.util.List;
+import javax.inject.Singleton;
+
+@Singleton public class NotifyOpponentFinishedUseCase {
```
[the remark this answers is not in the document]
Comment (nnelas) [reply 2048/2048/src/main/java/com/appcoins/eskills2048/usecase/NotifyOpponentFinishedUseCase.java:10]: didn't knew about that, changing

**Pull request merged**